src/Core/Migrate/Migrate.php

```diff
@@ -55,7 +57,7 @@ public static function migrateSettingsFile(Event $event = null)
                     'mailer_host'       => '127.0.0.1',
                     'mailer_user'       => '~',
                     'mailer_password'   => '~',
-                )
+                ) + $default_parameters['parameters']
```

Comment:
@Shine-neko : because with array_merge the parameters defined in $default_parameters would overwrite the legacy values :)